api/incubator/src/main/java/io/opentelemetry/api/incubator/logs/ExtendedLogger.java

```diff
@@ -41,6 +44,130 @@ default boolean isEnabled() {
     return isEnabled(Severity.UNDEFINED_SEVERITY_NUMBER);
   }
 
+  // Fluent log API
+
+  /**
+   * A fluent log event API, with convenience methods for incrementally adding fields and
+   * attributes.
+   *
+   * <p>Callers must call {@link LogEventBuilder#emit()}.
+   */
+  LogEventBuilder logBuilder(Severity severity, String eventName);
```

Comment:
callback methods are popular to short circuit if the severity is not reached, e.g. https://logging.apache.org/log4j/2.x/javadoc/log4j-api/org/apache/logging/log4j/Logger.html#info(java.lang.String,org.apache.logging.log4j.util.Supplier...)

we could add

```java
void log(Severity severity, Consumer<LogEventBuilder> consumer);
```